prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml

```diff
@@ -233,56 +94,40 @@ period_2:
       vars:
         periodEnd: XAD.14
 
-extension_9-effectiveDate:
-  generateList: true
-  expressionType: nested
-  vars:
-    periodStart: XAD.13
-  condition: $periodStart NOT_NULL
-  expressionsMap:
-    url:
-      type: SYSTEM_URL
-      value: 'xad13-effective-date'
-    valueString:
-      type: STRING
-      valueOf: $periodStart
-      expressionType: HL7Spec
-
-extension_10-expirationDate:
+extension:
   generateList: true
   expressionType: nested
-  vars:
-    periodEnd: XAD.14
-  condition: $periodEnd NOT_NULL
-  expressionsMap:
-    url:
-      type: SYSTEM_URL
-      value: 'xad14-expiration-date'
-    valueString:
-      type: STRING
-      valueOf: $periodEnd
-      expressionType: HL7Spec
-
-extension_8-xad19Addressee:
-  generateList: true
-  expressionType: nested
-  expressionsMap:
-    url:
-      type: SYSTEM_URL
-      value: xad19-addressee
-    valueCode:
-      type: STRING
-      valueOf: XAD.19
-      expressionType: HL7Spec
+  expressions:
+    - expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: iso21090-ADXP-censusTract
+        valueCode:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: XAD.10
+    - vars:
+        xad7: String, XAD.7
+      condition: $xad7 EQUALS HV
+      expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: iso21090-AD-use
+        valueCode:
+          type: STRING
+          valueOf: XAD.7
+          expressionType: HL7Spec
+    - expressionType: nested
+      condition: $hl7v2Name NOT_NULL
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: hl7v2Name
+        valueString:
+          type: STRING
+          valueOf: $hl7v2Name
+    - expressionType: resource
```

Comment:
Looking for feedback here.  I opted not to do a null check against field here on the basis of
- something is going to be populated
- if nothing is populated, it just will be a url with an empty `extension` array

Reply:
I'm not sure what you mean here. Are you talking about a null check on the expression with `iso21090-ADXP-censusTract` for the url?

Reply:
So if literally none of the values that extension map are present, there will be an empty extension.  i.e. I'm not writing
`xad.1 NULL || xad.2 NULL || xad.3 NULL ...`

Reply:
Gotcha. Yes, I think that's okay. Does the XAD.10 expression need a null check?

Reply:
Yep... missing from the initial implementation.  Will fix it now